lib/components/molecules/AtomicCarousel/AtomicCarousel.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+/**
+ *
+ * AtomicCarousel
+ *
+ */
+import React, { useState, useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+
+import styles from './AtomicCarousel.style';
+import type { Props } from './types';
+import List from '../List';
+import ListItem from '../../atoms/ListItem';
+import { ButtonVanilla } from '../../atoms/Button';
+
+// Rendering all dots as button
+const getSliderDots = (slidesCount, currentItem, setItem) => {
+  const specificNode = useCallback(index => setItem(index));
+  const countElements = [];
+
+  for (let i = 1; i <= slidesCount; i += 1) {
+    countElements.push(
+      <ListItem key={i} className="atomic-item">
+        <ButtonVanilla disabled={currentItem === i} onClick={() => specificNode(i)}>
+          {i}
+        </ButtonVanilla>
+      </ListItem>
+    );
+  }
+
+  return countElements;
+};
+
+// Calculating required number of dots
+const getList = (items, setItem, currentItem, perSlideFrame) => {
+  const slidesCount = Math.ceil(items.length / perSlideFrame);
+
+  return (
+    <div className="atomic-dots">
+      <List>{getSliderDots(slidesCount, currentItem, setItem)}</List>
+    </div>
+  );
+};
+
+// Main Component
+const AtomicCarousel = ({
+  dots,
+  navigation,
+  children,
+  className,
+  perSlideFrame,
+  prevLabel,
+  nextLabel,
+}: Props): Node => {
+  const carouselClasses = [...className.split(','), 'atomic-carousel'].join(' ');
+  const [currentSlide, setCurrentSlide] = useState(1);
+  const [data, setData] = useState([]);
```

Comment:
What does data and setData mean? What does it represent

Reply:
data is the actual slider content as this component is using children prop so it could be a single element or multiple elements and when there is a single element child return object and when there are multiple it returns the array,

So to make the consistency I have used data state and convert children into this array.